assembly/src/parsers/blocks.rs

```diff
@@ -251,22 +254,45 @@ impl BlockParser {
 // ================================================================================================
 
 pub fn combine_blocks(mut blocks: Vec<CodeBlock>) -> CodeBlock {
-    // merge consecutive Span blocks
+    // merge consecutive Span blocks.
     let mut merged_blocks: Vec<CodeBlock> = Vec::with_capacity(blocks.len());
+    // Keep track of all the consecutive Span blocks and are merged together when there is a discontinuity.
+    let mut continuous_span_vector: Vec<Span> = Vec::new();
```

Comment:
minor nit: I wouldn't put the type of a variable into the suffix. Maybe `contiguous_spans` would work better here?